hooks/useLogin.ts

```diff
@@ -18,27 +24,30 @@ export enum STATUS {
 // {issuer, publicAddress, email}
 // https://github.com/magiclabs/magic-js/blob/master/packages/%40magic-sdk/types/src/modules/user-types.ts#L17-L21
 
-// const $metadata = useLogin('/go-somewhere-if-it-does-not-work')
+// const $metadata = useLogin({redirect: '/go-somewhere-if-it-does-not-work'})
 
 export interface LoginProps {
   redirect?: string
-  timeout?: number
 }
+
 export function useLogin(config: LoginProps = {}) {
-  const { redirect, timeout = -1 } = config
+  const $router = useRouter()
+  const { redirect } = config
   const [$status, $setStatus] = useState<STATUS>(STATUS.LOADING)
   const [$error, $setError] = useState<ApiError | LocalError | null>(null)
   const [$magicMetadata, $setMagicMetadata] =
     useState<MagicUserMetadata | null>(null)
   const [$metadata, $setMetadata] = useState<ApiUserMetadata | null>(null)
+
   useEffect(() => {
     const checkLoggedIn = async () => {
       try {
-        // this is likely a case where we're working in not-the-browser
-        if ($metadata || !magic || !magic.user) {
-          Promise.reject(new LocalError('Magic instance not available!', 500))
+        // if we're already loaded, quit
+        if ($status === STATUS.LOADED) return
```

Comment:
I don't think so -- This `useEffect` runs every time one of its dependencies is updated, and it calls `checkLoggedIn` every time it re-runs. `$status` is one of its dependencies, so for example, in the case where a user does not have a valid login token:

1. `checkLoggedIn` will execute on page load
1. `token` will not be set
1. Control flow proceeds down the `!token` path and calls `$setStatus(STATUS.NOT_FOUND)`
1. Updating the status then triggers the `useEffect` to run again, which will call `checkLoggedIn` again
1. `checkLoggedIn` will make another call to Magic that proceeds down the same control flow as the prior useEffect execution
1. Since the status is set to its now-current value, NOT_FOUND, the useEffect does not run again.

We're still making unnecessary network requests as a result of our state + effect design, which is why the `$status` + `useEffect` design needs to be considered more, but I think it's fine for now since I haven't seen a situation where it has a user-visible effect.